### PR TITLE
Add support for port = 0 in Redis driver

### PR DESCRIPTION
**Pull request by MrPetovan**
Related to #13318 

- Add support for Unix domain sockets to Redis in setting defaults

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -77,6 +77,12 @@
 		"npm-asset/textcomplete": "^0.18.2",
 		"npm-asset/typeahead.js": "^0.11.1"
 	},
+	"suggest": {
```
Comment (nupplaphil) [inline composer.json:80]: Cool!! I didn't know about that option

**Review by nupplaphil — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (MrPetovan) [reply composer.json:80]: It suppresses some of the "missing `ext-*` in composer.json file" warnings that PhpStorm is displaying!

Comment (nupplaphil) [reply composer.json:80]: It annoyed me all the time with phpstorm too.. thanks for that ;-)

File: src/Core/Cache/Type/RedisCache.php [hunk 2]
```diff
@@ -57,22 +57,27 @@ public function __construct(string $hostname, IManageConfigValues $config)
...
-		$redis_db   = $config->get('system', 'redis_db', 0);
+		$redis_db   = (int)$config->get('system', 'redis_db', 0);
 
 		try {
-
-			if (!empty($redis_port) && !@$this->redis->connect($redis_host, $redis_port)) {
-				throw new CachePersistenceException('Expected Redis server at ' . $redis_host . ':' . $redis_port . ' isn\'t available');
-			} else if (!@$this->redis->connect($redis_host)) {
-				throw new CachePersistenceException('Expected Redis server at ' . $redis_host . ' isn\'t available');
+			if (is_numeric($redis_port) && $redis_port > -1) {
```
Comment (nupplaphil) [inline src/Core/Cache/Type/RedisCache.php:63]: So this will include `0` as well, wouldn't it?

Can you add a test to phpunit as well?

**Review by nupplaphil — CHANGES_REQUESTED** [review]

File: src/Core/Cache/Type/RedisCache.php (hunk 2 above)
Comment (MrPetovan) [reply src/Core/Cache/Type/RedisCache.php:63]: I'm sorry, I looked at the `src/Core/Cache/RedisCacheTest.php` file and I'm not sure what I should add or change to test the port = 0 case.

Comment (nupplaphil) [reply src/Core/Cache/Type/RedisCache.php:63]: I'll try to add it when I'm back from holiday ;-))
--> merge it meanwhile

**Pull request merged**